internal/bindings/server.go

```diff
@@ -291,6 +301,9 @@ func connectWithDial(handle C.uintptr_t, address *C.char, fd *C.int) C.int {
 	return C.int(0)
 }
 
+// Context of the parent node to be used by the dial function.
+var dialCtx context.Context
```

Comment:
This seems wrong to being a package level var in case there are multiple instances of the dqlite client.

Reply:
This was it! I didn't have a good grasp of how the server was starting up and how the dqlite C bindings talked to the rest of the Go code. The tests failed when a node went offline and I thought this was due to the cluster losing quorum but it was just this variable causing the wrong context to be cancelled.